### PR TITLE
Fix comment in Zip2

### PR DESCRIPTION
**Pull request by PatrickPijnappel**
If we exhaust the second sequence first, that means the first sequence is longer.

### CODE REVIEW

File: stdlib/public/core/Zip.swift [hunk 1]
```diff
@@ -38,11 +38,11 @@ public struct Zip2Generator<
   ///   since the copy was made, and no preceding call to `self.next()`
   ///   has returned `nil`.
   public mutating func next() -> Element? {
-    // The next() function needs to track if it has reached the end.  If we
-    // didn't, and the first sequence is shorter than the second, then, when we
+    // The next() function needs to track if it has reached the end. If we
+    // didn't, and the first sequence is longer than the second, then when we
```
Comment (moiseev) [inline stdlib/public/core/Zip.swift]: Good catch!

File: stdlib/public/core/Zip.swift [hunk 2]
```diff
@@ -38,11 +38,11 @@ public struct Zip2Generator<
...
     // have already exhausted the second sequence, on every subsequent call to
     // next() we would consume and discard one additional element from the
-    // first sequence, even though next() return nil.
+    // first sequence, even though next() is already returning nil.
```
Comment (moiseev) [inline stdlib/public/core/Zip.swift]: Can't quite agree on this one. Perhaps something like 'would return'?

Comment (PatrickPijnappel) [inline stdlib/public/core/Zip.swift]: How about 'even though we return nil'

Comment (jtbandes) [inline stdlib/public/core/Zip.swift]: I'd suggest "even though next() had already returned nil`.

Comment (moiseev) [inline stdlib/public/core/Zip.swift]: @jtbandes :+1:

Comment (PatrickPijnappel) [inline stdlib/public/core/Zip.swift]: @moiseev Modified.

File: stdlib/public/core/Zip.swift (hunk 1 above)
Comment (parinyaprasatsan) [inline stdlib/public/core/Zip.swift]: Diff

**Pull request merged**